### PR TITLE
 CSIC-50: mount and unmount of treeq

### PR DESCRIPTION
**Pull request by gadekarnitesh**
Changes:

1.mount and unmount treeq volume to pod

### CODE REVIEW

File: deploy/helm/infinibox-csi-driver/values.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ images:
   provisionersidecar: quay.io/k8scsi/csi-provisioner:v1.4.0
 
   # "images.snapshotter-sidercar" defines the container image used for the csi snapshotter sidercar
-  snapshottersidecar: quay.io/k8scsi/csi-snapshotter:v1.2.2
+  snapshottersidecar: quay.io/k8scsi/csi-snapshotter:v2.0.1
```
Comment (Nilesh20) [inline deploy/helm/infinibox-csi-driver/values.yaml]: Why are you changing this version. Is it needed for treeQ.

File: storage/nfscontroller.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func getPermission(permission string) ([]map[string]interface{}, error) {
 	var permissionMap []map[string]interface{}
 	err := json.Unmarshal([]byte(result), &permissionMap)
 	if err != nil {
-		log.Errorf("invalid nfs_export_permissions formate", err)
+		log.Errorf("invalid nfs_export_permissions formate %v", err)
```
Comment (Nilesh20) [inline storage/nfscontroller.go]: Format spelling

**Review by Nilesh20 — CHANGES_REQUESTED** [review]

File: deploy/helm/infinibox-csi-driver/values.yaml (hunk 1 above)
Comment (gadekarnitesh) [reply deploy/helm/infinibox-csi-driver/values.yaml]: changes reverted . it's not required for treeq

**Review by Nilesh20 — APPROVED** [review]

**Review by Nilesh20 — APPROVED** [review]

File: api/nfsapiclient.go [hunk 3]
```diff
@@ -295,7 +295,7 @@ func (c *ClientService) AddNodeInExport(exportID int, access string, noRootSquas
 		}
 	}
 	if index != -1 {
-		permissionList = removeIndex(permissionList, index)
+		//permissionList = removeIndex(permissionList, index)
```
Comment (dhanashreebiradar) [inline api/nfsapiclient.go]: Please remove comment

File: api/treeqclient.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+//CreateTreeq method create treeq
+func (c *ClientService) CreateTreeq(filesystemID int64, treeqParameter map[string]interface{}) (*Treeq, error) {
+	var err error
+	defer func() {
+		if res := recover(); res != nil && err == nil {
+			err = errors.New("CreateTreeq Panic occured -  " + fmt.Sprint(res))
+		}
+	}()
+	log.Info("Create filesystem")
```
Comment (dhanashreebiradar) [inline api/treeqclient.go:102]: It should be Craete Treeq

**Review by dhanashreebiradar — COMMENTED** [review]
Please fix comments

**Review by dhanashreebiradar — APPROVED** [review]
Lg

**Pull request merged**